### PR TITLE
Delete command supports multiple index deletions

### LINKED ISSUE #59: Allow deleting multiple people with one command

### PR DESCRIPTION
**Pull request by ChillinRage**
Fix #59

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/index/IndexList.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+        return true;
+    }
+
+    @Override
+    public String toString() {
+        String lineOfIndices = "";
+        for (Index index: indices) {
+            lineOfIndices += index.toString() + " ";
+        }
```
Comment (E0735389) [inline src/main/java/seedu/address/commons/core/index/IndexList.java:89]: This sounds like exactly the wrong way to use string builder (O(n^2) performance). (But of course in practice it won't matter)

File: src/main/java/seedu/address/commons/core/index/IndexList.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
+package seedu.address.commons.core.index;
+
+import java.util.ArrayList;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Represents an array of zero-based or one-based indices maintained in descending order.
```
Comment (E0735389) [inline src/main/java/seedu/address/commons/core/index/IndexList.java:8]: The docstring doesn't match the code?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -38,6 +39,24 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
         return Index.fromOneBased(Integer.parseInt(trimmedIndex));
     }
 
+    /**
+     * Parses {@code args} into an array of {@code Index} and returns it in ascending order.
```
Comment (E0735389) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:43]: This docstring also doesn't match the code.

**Review by E0735389 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/commons/core/index/IndexList.java (hunk 1 above)
Comment (ChillinRage) [reply src/main/java/seedu/address/commons/core/index/IndexList.java:89]: yep it's inefficient. I don't really know a more OOP-friendly way to build the string.
Realistically, most users wouldn't use delete for large number of indices.

File: src/main/java/seedu/address/commons/core/index/IndexList.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    /**
+     * Returns the current size of the index list.
+     *
+     * @return integer representing the number of elements in the list.
+     */
+    public int size() {
+        return indices.size();
+    }
+
+    public Index getIndex(int index) {
```
Comment (E0735389) [inline src/main/java/seedu/address/commons/core/index/IndexList.java:48]: The name of this method seems excessively confusing (both the index into the list, and the returning index, are "index").

File: src/main/java/seedu/address/commons/core/index/IndexList.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            indices.add(index);
+        }
+    }
+
+    /**
+     * Returns the current size of the index list.
+     *
+     * @return integer representing the number of elements in the list.
+     */
+    public int size() {
```
Comment (E0735389) [inline src/main/java/seedu/address/commons/core/index/IndexList.java:44]: Method name must be verb. (But to be fair the underlying class name is also `size`)

File: src/main/java/seedu/address/commons/core/index/IndexList.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+     * IndexList can only be created by calling {@link IndexList#emptyList()}.
+     */
+    private IndexList() {
+        indices = new ArrayList<>();
+    }
+
+    /**
+     * Creates a new {@code IndexList} with an empty list.
+     */
+    public static IndexList emptyList() {
```
Comment (E0735389) [inline src/main/java/seedu/address/commons/core/index/IndexList.java:24]: Same comment as below.

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java [hunk 7]
```diff
@@ -57,7 +58,7 @@ public void execute_validIndexFilteredList_success() {
         DeleteCommand deleteCommand = new DeleteCommand(INDEX_FIRST_PERSON);
 
         String expectedMessage = String.format(DeleteCommand.MESSAGE_DELETE_PERSON_SUCCESS,
-                Messages.format(personToDelete));
+                Messages.format(personToDelete)) + "\n";
```
Comment (E0735389) [inline src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:61]: Why are we adding the `"\n"` here?

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 8]
```diff
@@ -5,44 +5,70 @@
...
- * Deletes a person identified using it's displayed index from the address book.
+ * Deletes 1 or more people identified using it's displayed index from the address book.
  */
 public class DeleteCommand extends Command {
 
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
+            + ": Deletes 1 or more people identified by their index numbers used in the displayed person list.\n"
```
Comment (E0735389) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:23]: From the description alone, it isn't clear how to use the command for multiple people. (comma separated, space separated, etc.?)

**Review by E0735389 — CHANGES_REQUESTED** [review]
Also remember to update UG.

If you want more features, you can also choose to implement `delete 1-3,5,7-end` (like how Microsoft Word's page range specification for printer etc. work)

**Comment by ChillinRage** [comment]
Will update UG later

**Review by E0735389 — APPROVED** [review]
Apart from UG, looks good.

**Review by E0735389 — CHANGES_REQUESTED** [review]
* It would makes sense to make `delete 1,1` raises an error instead of silently accept.
* Remember to update UG.
* Also update the inline example in the code. The example is just `delete 1` at the moment, which does not explain how to delete multiple indices.

**Review by E0735389 — APPROVED** [review]

**Pull request merged**